### ISSUE TITLE
DBADASH Removing a server with powershell is not removing it from GUI

### ISSUE THREAD

**Issue by aachaemenes** [trimble-oss/dba-dash#995]
Have the latest version and tried to remove a server using the powershell. the server name does not show in Json and under source when I search for it but still shows under Snapshot Age alert and the list of servers in GUI.

**Comment by DavidWiseman** [comment]
In the GUI, select *Options\Manage Instances*.  Click the **Mark Deleted** button to mark the instance as deleted.

[See here](https://dbadash.com/docs/help/faq/#how-do-i-remove-an-instance) for more info.

**Comment by aachaemenes** [comment]
Any way we can automate this through PowerShell?

**Comment by DavidWiseman** [comment]
You can use this as a starting point (using dbatools):

```pwsh
$RepositoryDBInstance = 'LOCALHOST'
$RepositoryDBName = 'DBADashDB'
$InstanceToDelete = 'NAME_OF_INSTANCE'

$InstanceID = Invoke-DbaQuery -SqlInstance $RepositoryDBInstance -Database $RepositoryDBName  -Query "SELECT InstanceID FROM dbo.Instances WHERE ConnectionID = @ConnectionID" -SqlParameter @{ConnectionID=$InstanceToDelete} -As SingleValue

if($InstanceID -ne $null){
    Invoke-DbaQuery -SqlInstance $RepositoryDBInstance -Database $RepositoryDBName = -Query "dbo.Instance_Del @InstanceID=@InstanceID" -SqlParameter @{InstanceID=$InstanceID} 
    Write-Host "Instance marked deleted" -ForegroundColor Magenta
}
else{
    Write-Host "Instance not found" -ForegroundColor Red
}
```

The instance is marked as deleted and is no longer displayed in the GUI.  This just sets IsActive=0 in the dbo.Instances table.  There is a @HardDelete=1 parameter that can be used to delete the associated data but there is a check to ensure that the instance is first marked IsActive=0 and there needs to be at least 24 hours passed since the last collection.

Hope this helps.

**Comment by aachaemenes** [comment]
this deleting business is becoming a real pain. can we have one script that does it all. meaning when we run it it deletes it from the source and json, then deletes it from GUI.

And a second script to purge the data from the DB after 24 hours.

**Comment by DavidWiseman** [comment]
It should be possible to automate the whole thing with the PS script provided, but I think it can be made a little easier.

For hard deleting an instance I'm going to look at adding an option to automatically do this after a specified number of days.  If you hard delete an instance immediately there is more work to do.  If you leave it for a while, much of the data will already be purged (using truncate) based on the data retention policies and there will be a smaller delete operation.  Also, sometimes it can be useful to have access to the data collected from a deleted instance for a period of time. Either way it would be good to have the data removed automatically after whatever period of time.

I think the CLI could be updated to optionally delete the instance from the repository DB.

**Comment by aachaemenes** [comment]
Thanks a million. It's much needed

**Comment by DavidWiseman** [comment]
This should be improved in the next release.  Deleting an instance in the config tool will now give you the option to mark deleted in the repository.  In the GUI, instances can now be deleted using a context menu in the tree.  I've also added a recycle bin folder where you can restore instances & fully delete them.  
The option to automatically hard delete instances can also be configured in the GUI.

**Comment by aachaemenes** [comment]
Thanks a million.

**Comment by DavidWiseman** [comment]
Included in 3.12 🚀
